src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/courtsentencing/CourtSentencingService.kt

```diff
@@ -53,18 +55,24 @@ class CourtSentencingService(
         val nomisResponse =
           nomisApiService.createCourtCase(offenderNo, courtCase.toNomisCourtCase())
 
+        val nomisCourtAppearanceResponse = nomisResponse.courtAppearanceIds.first()
         CourtCaseAllMappingDto(
           nomisCourtCaseId = nomisResponse.id,
           dpsCourtCaseId = courtCaseId,
           // expecting a court case with 1 court appearance - separate event for subsequent appearances
           courtAppearances = listOf(
             CourtAppearanceMappingDto(
               dpsCourtAppearanceId = courtCase.latestAppearance.appearanceUuid.toString(),
-              nomisCourtAppearanceId = nomisResponse.courtAppearanceIds.first().id,
-              nomisNextCourtAppearanceId = nomisResponse.courtAppearanceIds.first().nextCourtAppearanceId,
+              nomisCourtAppearanceId = nomisCourtAppearanceResponse.id,
+              nomisNextCourtAppearanceId = nomisCourtAppearanceResponse.nextCourtAppearanceId,
             ),
           ),
-          courtCharges = listOf(),
+          courtCharges = courtCase.latestAppearance.charges.mapIndexed { index, dpsCharge ->
```

Comment:
this is all dependent on nomis returning the charge Ids in the order they were sent to be created